espressif/vscode-esp-idf-extension
# Enhance/debug pause breakpoint

## Description

Pause program execution when trying to set a breakpoint during execution

Fixes #958

## Type of change

Please delete options that are not relevant.
- Bug fix (non-breaking change which fixes an issue)

## Steps to test this pull request

Provide a list of steps to test changes in this PR and required output

1. Click on F5 to start debugging a esp-idf application
2. When the program is running, try to set a breakpoint. The program will pause.
3. Observe results.

- Expected behaviour:

After breakpoint is set when program is running, the program execution should be paused and the breakpoint is set.

- Expected output:

## How has this been tested?

Manual testing using blink example.

**Test Configuration**:
* ESP-IDF Version: 5.0
* OS (Windows,Linux and macOS): macOS

## Dependent components impacted by this PR:

- ESP Debug Adapter

## Checklist
